### PR TITLE
[WIP] Full removal of movement_delay(), plus assorted refactors.

### PR DESCRIPTION
**Pull request by silicons**
- [x] Port carbon movement_delay()
- [ ] Port human species movement_delay and everything under it
- [x] Port flying movetype to traits for better management of movetypes
- [x] Implement new movespeed modifier flags

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -920,3 +922,23 @@
 /mob/setMovetype(newval)
 	. = ..()
 	update_movespeed(FALSE)
+
+/mob/add_trait(id)
+	. = ..()
+	if(.)
+		update_movespeed()
+		if(id == TRAIT_FLIGHT)
+			update_movement_type()
```
Comment (XDTM) [inline code/modules/mob/mob.dm]: i keep having to remove those

stop snowflaking the add_trait proc, use wrappers

Comment (kevinz000) [reply code/modules/mob/mob.dm]: can we just get an on_trait_add proc or nah

Comment (XDTM) [reply code/modules/mob/mob.dm]: You're probably better off using an `add_flight` and `remove_flight` proc, check out the fakedeath one if you want an example.
Alternatively if you want to condense it into one make a `set_flight(TRUE/FALSE, source)`.

I'm trying to keep traits as close as possible to read-only "flags", or the snowflake code will pile up rapidly if more systems try to check for a specific trait update to call their own proc.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -176,13 +176,15 @@
...
 
-/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/movement_delay()
-	var/turf/T = get_turf(src)
-	if(locate(/obj/structure/spider/stickyweb) in T)
-		speed = 2
-	else
-		speed = 7
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/Cross(atom/AM)
+	. = ..()
+	if(istype(/obj/structure/spider/stickyweb))
```
Comment (vuonojenmustaturska) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: you forgot something here

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -176,13 +176,15 @@
...
+		add_movespeed_modifier(MOVESPEED_MODIFIER_TARANTULA_WEB, multiplicative_slowdown = -5)
+
+/obj/living/simple_animal/hostile/poison/giant_spider/tarantula/Uncross(atom/AM)
 	. = ..()
+	if(istype(/obj/structure/spider/stickyweb))
```
Comment (vuonojenmustaturska) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: ditto

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 2 above)
Comment (kevinz000) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: fuck lmao

**Comment by optimumtact** [comment]
Closed until you finish #40428

**Pull request closed without merging**